tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -118,27 +113,23 @@ private void Dispose(bool disposing)
             }
         }
 
-        private ClusterType MapToplogyToClusterType(BsonValue topology)
-        {
-            switch (topology.AsString)
-            {
-                case "single": return ClusterType.Standalone;
-                case "replicaset": return ClusterType.ReplicaSet;
-                case "sharded": return ClusterType.Sharded;
-                default: throw new ArgumentException("Unknown topology type");
-            }
-        }
-
         private void Run(BsonDocument shared, BsonDocument test)
         {
             if (test.Contains("skipReason"))
             {
                 throw new SkipException($"Test skipped because {test["skipReason"]}.");
             }
 
-            if (shared.TryGetValue("topology", out var topology))
+            CheckTransactionFeature();
+
+            if (shared.TryGetValue("runOn", out var runOn) || shared.TryGetElement("topology", out var topology))
             {
-                CheckClusterTopology(topology.AsBsonArray);
+                var runOnArray =
+                    runOn != null
+                        ? runOn.AsBsonArray
+                        : new BsonArray(new[] { new BsonDocument(topology.Name, topology.Value) });
+
+                RequireServer.Check().RunOn(runOnArray);
```

Comment:
This seems too complicated where you are trying to simulate runOn from topology.

Why not just leave the original code for "topology" as it is and simply add this code:

    if (shared.TryGetValue("runOn", out var runOn))
    {
        RequireServer.Check().RunOn(runOn.AsBsonArray);
    }

Reply:
Because `RunOn` already has validation on `topology`. As I understand `runOn` is just a newer version of checking on `topology`, so I don't see why we need to have both validations about the same thing

Reply:
The question here that in case if we leave it as you propose we will have `code duplication`  indefinitely until we will finally update all test files. Everybody who works with this code will need to consider two code blocks for the same target. It cannot be good.

>Then why don't we remove support for "topology"?

the answer is because we still have JSON tests in old format, as soon as we will not have them, we will be able to simply remove mapping a `topology` format on `runOn`.

Reply:
There is no real code duplication... the code for "topology" and the code for "runOn" are different.

The code for "topology" already exists and doesn't need to be changed. We just need a tiny change to support "runOn" (the code I suggested above).

Reply:
You should add a comment somewhere that support for "topology" will be removed soon when the JSON files are all updated to the newer format that no longer uses "topology".